pkg/operator/sync.go

```diff
@@ -204,7 +212,7 @@ func (optr *Operator) syncBaremetalControllers(config *OperatorConfig) error {
 		return err
 	}
 
-	metal3Deployment := newMetal3Deployment(config, baremetalProvisioningConfig)
+	metal3Deployment := newMetal3Deployment(config, *baremetalProvisioningConfig)
```

Comment:
It doesn't make great sense to dereference the pointer here, maybe pass the address to newMetal3Deployment and dereference where it's actually used?  That would require re-checking non-nil, though (to be safe, anyway).  Anyway, just a suggestion.

Reply:
Yes, I am working on some improvements in baremetal_pod.go that includes what you just suggested.